### ISSUE TITLE
Docker compose run on every config apply with extends

### ISSUE THREAD

**Issue by luckyraul** [puppetlabs/puppetlabs-docker#226]
## What you expected to happen?
one run only
## What happened?

docker-compose runs every config apply

https://github.com/puppetlabs/puppetlabs-docker/blob/master/lib/puppet/provider/docker_compose/ruby.rb#L23 

returns array with 0 containers because image in empty

## How to reproduce it?

```
version: '2'
services:
  base:
    image: someimage

  web:
    restart: unless-stopped
    extends: base

  cron:
    restart: unless-stopped
    extends: base
    command: run cron
```

```
docker_compose { '/somepath/docker-compose.yml':
  ensure  => present,
}
```
## Versions:

```
$ puppet --version

$ docker version

$ facter os

$ puppet module list
```

## Logs:
```
Paste any relevant logs from a puppet run
```

**Comment by scotty-c** [comment]
@luckyraul can we have more information please, like that logs of the catalogue run

**Comment by davejrt** [comment]
@luckyraul as per our template, it helps us a lot if you provide a manifest file and a working template to work with. Please provide the information requested in the template so we can triage the issue

**Comment by luckyraul** [comment]
@davejrt updated in the description

**Comment by davejrt** [comment]
What error message are you getting? Can you please post the log files, or the lines of the log files you're concerned about?

**Comment by luckyraul** [comment]
@davejrt It applies on every run

**Comment by davejrt** [comment]
Please provide a log file with and without using extends showing the issue

**Comment by luckyraul** [comment]
@davejrt This is a quite obvious problem. Just look at the code which I pointed out. When there is a new run, It collects all images to hash for each service, and then try to compare them with actually used in containers. There is no image tag in extended service - so it is failed to compare and mark service for reapplying. You don't need any logs to understand that.

**Comment by davejrt** [comment]
@luckyraul we’re a small team and we don’t have the resource capacity to test every issue. The point of filling out the template is to reduce the effort in triaging these issues and help us, help you. That way we can determine if we need to raise a ticket internally and dedicate a resource to the issue.

**Comment by luckyraul** [comment]
In this code #{array['image']} is empty
```
Hash[*compose_file['services'].each.map { |key, array|
                 Puppet.info("Checking for compose service #{key} #{array['image']}")
                 ["#{key}-#{array['image']}", containers.count("#{key}-#{array['image']}")]
}.flatten]
```

You should check for extends and try to use parent image. I wrote some not tested code to fix the issue

```
image = array['extends'].nil? ? array['image'] : compose_file['services'][array['extends']]['image']
["#{key}-#{image}", containers.count("#{key}-#{image}")]
```

**Comment by luckyraul** [comment]
If I had enough free time I would not create this issue instead of a pull request. 
Now I can only point you directly to issue and make a suggestion how to fix that.

**Comment by davejrt** [comment]
fixed in #288
